calico/felix/fetcd.py

```diff
@@ -230,13 +311,37 @@ def start_watch(self, splitter):
     @actor_message()
     def force_resync(self, reason="unknown"):
         """
-        Force a resync from etcd after the current poll completes.
+        Force a resync with etcd after the current poll completes.
 
         :param str reason: Optional reason to log out.
         """
-        _log.info("Forcing a resync from etcd.  Reason: %s.", reason)
+        _log.info("Forcing a resync with etcd.  Reason: %s.", reason)
         self._watcher.resync_after_current_poll = True
 
+        if self._config.REPORT_ENDPOINT_STATUS:
+            _log.info("Endpoint status reporting enabled, marking existing "
+                      "endpoints as dirty so they'll be resynced.")
+            ep_ids = self._endpoint_status.keys()  # Avoid race with update.
+            for ep_id in ep_ids:
+                self.mark_endpoint_dirty(ep_id)
+
+    @actor_message()
+    def on_endpoint_status_changed(self, endpoint_id, status):
+        if not self._config.REPORT_ENDPOINT_STATUS:
+            _log.debug("Endpoint status reporting disabled, ignoring report.")
+            return
+        # We directly update the shared data-structure, which is safe as long
+        # as the reporting thread only does exact reads (and not iteration,
+        # say).
```

Comment:
This sort of comment always makes me nervous, even though I think it is correct. However since we are using greenlets it's fine even if the comment is wrong, and don't we rely on greenlets and explicit yields only in other places (thus making the comment unnecessary)?